### DISCUSSION TITLE
添加remote ruleset 就自动停止运行。

### DISCUSSION THREAD

**Discussion by imken** [muink/luci-app-homeproxy#18] [Q&A]
添加了几个ruleset, 来源是：
https://github.com/shangguanhongxin/for-sing-box-and-surge

添加完成后 homeproxy 启动后自动停止， 快速刷新后几秒内又再次停止，在“服务状态”中查看日志没有错误。
json 和 二进制 都是一样的情况。

检查json文件格式正确，例如
```
https://github.com/shangguanhongxin/for-sing-box-and-surge/raw/master/sing-box1.9/XiaoMi/XiaoMi.json
```

请问是否有 ruleset 数量限制？因此导致这样的错误。

另外请问 这些远程 ruleset 会保存到本地么？我没有找到，所以也无法判断是否正确读取远程ruleset

望解答，非常感谢。

**Comment by muink** [comment]
我测试是没问题的这个json
其他的请自行用 `sing-box rule-set format XiaoMi.json` 命令检查格式

**Comment by imken** [comment]
ImmortalWrt 23.05.3 r27917-81a1f98d5b
 
luci-app-homeproxy | git-24.226.23824-9a48dd3


有点奇怪，貌似不是内容格式问题，我自己建了一个测试的rule一样出错。

```
{
  "version": 1,
  "rules": [
    {
      "domain": [
        "lf3-static.bytedns111doc.com"
      ],
      "domain_suffix":[
        "111openwrt.ai", "111vsean.net", "111immortalwrt.org", "111kyarucloud.moe"
      ],
      "ip_cidr": [
        "129.146.222.181/32"
      ]
    }
  ]
}
```

我现在有11条启用的ruleset，这条测试的是第12条，一旦启用 homeproxy 就会在启动几秒之后自动停止。

尝试禁用之前正常11条的其中几条，再启用第12条，也一样报错，看来也不是数量问题。

在正常启用的ruset中加入 XiaoMi.json 的内容，反而正常运行。
有点玄学。。。。

**Comment by imken** [comment]
试着用 raw.gitmirror.com 来 替换掉 raw.githubusercontent.com, 发现小米这个rule 可以了，以为找到问题所在了。结果再加个其他的 rule 又不行了…… 不知道怎么试了，能力不足，哈哈哈。

**Comment by muink** [comment]
那就是没配置好，rule下载不了
